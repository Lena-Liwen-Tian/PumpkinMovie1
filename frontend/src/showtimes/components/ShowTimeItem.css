
.box{
    background-color: white;
    margin-bottom:3%;
    margin-right:5%;
    margin-left:3%;
    padding:2%;
    display: block;
    width:150px;
    height:100%;
    border-radius: 6px;
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.5);
}
.mg{
    width:145px;
    height:180px;
    margin-left:auto;
    
}
 .card--title{
     font-size:10pt;
    margin-top:10%;
    text-align: center;
}

.content{
    font-size: 8pt;
    padding-left:5%;
    padding-right:5%;
    text-align: center;
}
.time{
    color:#FF7417;
}

