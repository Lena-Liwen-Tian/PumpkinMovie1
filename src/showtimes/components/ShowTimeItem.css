
.bu{
    width:20%;
    margin-bottom:10%;
    display: block;
    
}
.up{
    font-size: 18pt;
    margin-top:3%;
}
.card1 {
    margin-left:0;
    width:95%;
}
