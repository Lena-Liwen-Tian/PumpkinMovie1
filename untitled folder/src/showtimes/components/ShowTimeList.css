.u{    
    display: flex;
    overflow-x: scroll;
    background-color:#D0D0D0;

}
.con{
    padding-top:2%;
    padding-bottom:2%;
    display:block;
    margin-right:2%;
    

}
