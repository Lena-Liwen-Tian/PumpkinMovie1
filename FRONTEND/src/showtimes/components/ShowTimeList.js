import React from 'react';
import ShowTimeItem from './ShowTimeItem';
import './ShowTimeList.css'
const ShowTimeList = props =>{

        return (
          <span className="u">
            {props.items.map(time => (    
              <div className="con">       
              <ShowTimeItem
                key={time._id}
                date={time.date}
                imdbID={time.imdbID}
                theatre={time.theatre.name}
                theatreid={time.cinemaid}
                des = {time.des}
                image={time.newimage}
                title={time.title}
                time={time.time}
              />
              </div>     
            ))}
          </span>
        );
      };
      




export default ShowTimeList;



