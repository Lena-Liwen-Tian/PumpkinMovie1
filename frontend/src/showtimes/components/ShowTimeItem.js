import React, { useState, useContext } from 'react';
import { Link } from 'react-router-dom';
import Card from '../../shared/components/UIElements/Card';
import Button from '../../shared/components/FormElements/Button';

import './ShowTimeItem.css';


const ShowTimeItem = props => {
  let theatreid = props.theatreid.toString();
  return (
    <div className="box">
    <div className="row no-gutters">
      <div>  
      <img src= {props.image} className="mg" />
      </div>
      <div>
           <div >
           <Link to={`/description/${props.imdbID}`}><h6 className="card--title">{props.title}</h6></Link>
           <p className="content">{props.des}</p>
           <h6 className="card--title">Location</h6>
           <Link to={`/${theatreid}/theaterinfo`}><p className="content">{props.theatre}</p></Link>
           <h6 className="card--title">Time</h6>
           <h6 className="card--title">{props.time}</h6>
            <Link className="buy" to={`/checkout/${props.theatreid}/${props.imdbID}`}><h6 className="card--title">BUY</h6></Link>
            </div>
        </div>
      </div>
    </div>
    
   
  );
};

export default ShowTimeItem;







