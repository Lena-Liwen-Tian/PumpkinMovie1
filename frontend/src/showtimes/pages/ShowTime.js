import React, { useEffect,useState } from 'react';
import ShowTimeList from '../components/ShowTimeList';
import { useParams } from 'react-router-dom';
import Calendar from '../components/calendar';
import { useHttpClient } from '../../shared/hooks/http-hook';
import LoadingSpinner from '../../shared/components/UIElements/LoadingSpinner';
import axios from 'axios';
    
    const ShowTime = props => {
    
      const timeid = useParams().timeid;
      const [LoadedShowtimes, setLoadedShowtimes] = useState();
      const [isLoading,setisLoading] = useState(false);
      const [error, setError] = useState();
      useEffect(() => {
        
         const fetchShowtimes = async () => {
          setisLoading(true);

          try{          
            const response= await fetch(`https://pumpkintry.herokuapp.com/api/showtimes/${timeid}`); 
            const responseData = await response.json();
            
            if(!response.ok){
              throw new Error(responseData.message);
            }
            setLoadedShowtimes(responseData.showtimes);
            
          } catch (err) {
            setError(err.message);
          }
          setisLoading(false);
        };

        fetchShowtimes();
      },[timeid]);
      const errorHandler = () =>{
        setError=(null);
      }
     
      return (        
      <React.Fragment>
        <Calendar />
        {isLoading && (<div>
          <LoadingSpinner />
        </div>)}  
      {!isLoading && LoadedShowtimes &&<ShowTimeList items={LoadedShowtimes}/>}
      </React.Fragment>);
};



export default ShowTime;