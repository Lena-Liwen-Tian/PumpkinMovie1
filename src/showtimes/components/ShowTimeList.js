import React from 'react';
import ShowTimeItem from './ShowTimeItem';
import './ShowTimeList.css'
const ShowTimeList = props =>{
      

        return (
          <ul className="u">
            {props.items.map(time => (

              <ShowTimeItem
                key={time.id}
                id={time.id}
                image={time.image}
                title={time.name}
                rating={time.rating}
                time={time.showtimes}
              />
            ))}
          </ul>
        );
      };
      




export default ShowTimeList;



