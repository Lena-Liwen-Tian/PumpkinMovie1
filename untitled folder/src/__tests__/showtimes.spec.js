import React from 'react';
import { create } from "react-test-renderer";
import ShowTimeList from '../showtimes/components/ShowTimeList';
import ShowTimeItem from '../showtimes/components/ShowTimeItem';
import { mount,shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Weekday from '../showtimes/components/weekday';
import { Link, MemoryRouter } from 'react-router';
import Calendar from '../showtimes/components/calendar';
import ShowTime from '../showtimes/pages/ShowTime';
import toJson from 'enzyme-to-json';
import { act } from 'react-dom/test-utils';
import {render,unmountComponentAtNode} from 'react-dom';
import { FetchMock } from 'fetch-mock';
import {addWeeks,subWeeks,addMonths,subMonths,startOfWeek,addDays,format,isSameMonth,isSameDay,endOfWeek,startOfMonth,endOfMonth,parse} from 'date-fns';
configure({ adapter: new Adapter() });


   describe("test showtimesitem",()=>{
        it("should render 5 h6 showtimesitem",()=>{
            const showtimesitem =shallow(<ShowTimeItem date="2015-05-05" imdbID="345"
            theatre="Alamo" theatreid="1" des="good" image="app.jpg" title="Hulu" time="12"/>);     
            expect(showtimesitem.find("h6").length).toBe(5);        
        })
        it("should render 1 img showtimesitem",()=>{
            const showtimesitem =shallow(<ShowTimeItem date="2015-05-05" imdbID="345"
            theatre="Alamo" theatreid="1" des="good" image="app.jpg" title="Hulu" time="12"/>);     
            expect(showtimesitem.find("img").length).toBe(1);        
        })
        it("should have 3 links",()=>{
            const showtimesitem =shallow(<ShowTimeItem date="2015-05-05" imdbID="345"
            theatre="Alamo" theatreid="1" des="good" image="app.jpg" title="Hulu" time="12"/>);     
            expect(showtimesitem.find("Link").length).toBe(3);        
            expect(showtimesitem.find("Link").at(0).prop("to")).toBe("/description/345");
        })
    });
    describe("test showtimelist",()=>{
        it("timelist list rendered correctly",()=>{
            let LoadedShowtimes = [{
                key: "123",_id:"123",date:"2015-05-05",imdbID:"123",theatre:{name:"hello",theatreid:1},cinemaid:"1"
            }];
            const showtimelist= shallow(<ShowTimeList items={LoadedShowtimes}/>);
            expect(toJson(showtimelist)).toMatchSnapshot();
        })
        it("timelist list have only one item now",()=>{
            let LoadedShowtimes = [{
                key: "321",_id:"123",date:"2015-05-05",imdbID:"123",theatre:{name:"hello",theatreid:"1"},cinimaid:"1"
            }];
            const showtimelist= shallow(<ShowTimeList items={LoadedShowtimes}/>);
            expect(showtimelist.find("ShowTimeItem").length).toBe(1);
        })
    })
    describe("test showtime page",()=>{
        let container;
        beforeEach(()=>{
            container = document.createElement("div");
            document.body.appendChild(container);
        });
        afterEach(()=>{
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        })
    });
    describe("test Calender",()=>{
        it("should render Calendar correctly",()=>{
        let calendar = shallow(<Calendar />);
        expect(calendar.find("div").length).toBe(24);
        });

    });
    describe("test Showtimeitem",()=>{
        it("should render showtime item correctly",()=>{
        let showtimeitem = shallow(<ShowTimeItem key ="1" date="1" imdbID="1" theatre="1" theatreid="1" 
        des = "1" image="1" title="1" time="1"/>);
        expect(showtimeitem.find("h6").length).toBe(5);
        });
        it("should render showtime image correctly",()=>{
            let showtimeitem = shallow(<ShowTimeItem key ="1" date="1" imdbID="1" theatre="1" theatreid="1" 
            des = "1" image="1" title="1" time="1"/>);
            expect(showtimeitem.find("img").length).toBe(1);
            });
    

    });
    describe("test Weekday",()=>{
        it("should render weekday item correctly",()=>{
        let weekday = shallow(<Weekday />);
        expect(weekday.find("div").length).toBe(1);
        });
       
    });
  
