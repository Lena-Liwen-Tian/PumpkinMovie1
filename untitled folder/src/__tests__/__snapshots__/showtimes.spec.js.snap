// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test showtimelist timelist list rendered correctly 1`] = `
<span
  className="u"
>
  <div
    className="con"
  >
    <ShowTimeItem
      date="2015-05-05"
      imdbID="123"
      key="123"
      theatre="hello"
      theatreid="1"
    />
  </div>
</span>
`;
